server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/MailboxUpdateRequest.java

```diff
@@ -0,0 +1,155 @@
+        private Optional<SortOrder> sortOrder;
+
+        private Builder() {
+            id = Optional.empty();
+            name = Optional.empty();
+            role = Optional.empty();
+            sortOrder = Optional.empty();
+            parentId = Optional.empty();
+        }
+
+        public Builder id(String id) {
+            Preconditions.checkNotNull(id);
+            this.id = Optional.of(id);
+            return this;
+        }
+
+        public Builder name(String name) throws MailboxException {
+            Preconditions.checkNotNull(name);
+            this.name = Optional.of(name);
+            return this;
+        }
+
+        public Builder parentId(String parentId) {
+            this.parentId = Optional.ofNullable(parentId);
+            return this;
+        }
+
+        public Builder role(Role role) {
+            Preconditions.checkNotNull(role);
+            throw new NotImplementedException();
+        }
+
+        public Builder sortOrder(SortOrder sortOrder) {
+            Preconditions.checkNotNull(sortOrder);
+            throw new NotImplementedException();
+        }
+
+
+        public MailboxUpdateRequest build() {
+            return new MailboxUpdateRequest(id, name, parentId, role, sortOrder);
+        }
+    }
+
+    private final Optional<String> id;
+    private final Optional<String> name;
+    private final Optional<String> parentId;
+    private final Optional<Role> role;
+    private final Optional<SortOrder> sortOrder;
+
+    @VisibleForTesting
+    MailboxUpdateRequest(Optional<String> id, Optional<String> name, Optional<String> parentId, Optional<Role> role, Optional<SortOrder> sortOrder) {
+
+        this.id = id;
+        this.name = name;
+        this.parentId = parentId;
+        this.role = role;
+        this.sortOrder = sortOrder;
+    }
+
+    public Optional<String> getId() {
+        return id;
+    }
+
+    public Optional<String> getName() {
+        return name;
+    }
+
+    public Optional<String> getParentId() {
+        return parentId;
+    }
+
+    public Optional<Role> getRole() {
+        return role;
+    }
+
+    public Optional<SortOrder> getSortOrder() {
+        return sortOrder;
+    }
+
+
+    @Override
+    public final boolean equals(Object obj) {
+        if (obj instanceof MailboxUpdateRequest) {
+            MailboxUpdateRequest other = (MailboxUpdateRequest) obj;
+            return Objects.equals(this.id, other.id)
+                && Objects.equals(this.name, other.name)
+                && Objects.equals(this.parentId, other.parentId)
+                && Objects.equals(this.role, other.role)
+                && Objects.equals(this.sortOrder, other.sortOrder);
+        }
+        return false;
+    }
+
+    @Override
+    public final int hashCode() {
+        return Objects.hash(id, name, parentId, role, sortOrder);
+    }
+
+    @Override
+    public String toString() {
+        return com.google.common.base.Objects.toStringHelper(getClass())
```

Comment:
Now you can rebase on master :-) and use MoreObjects. Otherwise you will add eclipse deprecation warnings !